src/tunnel.cc

```diff
@@ -438,28 +438,13 @@ TunnelStateData::retryOrBail(const char *context)
 
     /* bail */
 
-    if (request)
-        request->hier.stopPeerClock(false);
-
     // TODO: Add sendSavedErrorOr(err_type type, Http::StatusCode, context).
     // Then, the remaining method code (below) should become the common part of
     // sendNewError() and sendSavedErrorOr(), used in "error detected" cases.
     if (!savedError)
         saveError(new ErrorState(ERR_CANNOT_FORWARD, Http::scInternalServerError, request.getRaw(), al));
```

Comment:
I think we should address (the essence of) this TODO in this PR because, at the very least, the introduction of "AndDestroy" logic affects this TODO -- sendSavedErrorOrAndDestroy() would not work well. We are also modifying at least one (out of two) code block related to this TODO.

AFAICT, the PR has two very similar "create a new error if none was saved" code snippets in sendErrorAndDestroy context right now: One here and one hidden in TunnelStateData::noteDestinationsEnd(). I would try to improve both use cases by moving the equivalent of this `if` statement into sendErrorAndDestroy() itself -- the method will create ErrorState if none was previously saved. If there are no other use cases, then the TODO will be fully addressed.

Please note that noteDestinationsEnd() would still check for nil savedError to set the `reason` parameter correctly. We could even convert the three sendErrorAndDestroy() calls there into one, but I would not do that because the resulting parameter calculations will look too complex.

Reply:
Done (9f09620).